### PR TITLE
Use the new cool Github templates

### PR DESCRIPTION
**Pull request by gpotter2**
You can now use Github[ issue templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms) to produce pretty cool forms. I thought those could be cool for the spack repo. (You can try those changes out on my repo: https://github.com/gpotter2/spack/issues/new/choose)

I have converted the existing templates with a minor tweaks:
- ~~removed the "please ping maintainers clause" because apparently you now have a bot that does this automatically (?)~~ (added. see the comment below)

You are free to provide feedback, modify this or not use it at all. It's a cool feature if you ask me, and it makes the issue templates much more readable.

Here are a few screenshots:

image
image

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]
Thanks. Waiting for other maintainers to have a look too.

**Comment by adamjstewart** [comment]
Bot only works for PRs, not issues yet

File: .github/ISSUE_TEMPLATE/build_error.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: "\U0001F4A5 Build error"
+description: Some package in Spack didn't build correctly
+title: "Installation issue: "
+labels: [build-error]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to report this build failure. To proceed with the report please:
+        1. Title the issue "Installation issue: <name-of-the-package>".
```
Comment (adamjstewart) [inline .github/ISSUE_TEMPLATE/build_error.yml]: It seems like `<name-of-the-package>` is being treated as an unknown HTML tag and not displaying. Let's remove the angle brackets and use italics instead?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/build_error.yml (hunk 1 above)
Comment (gpotter2) [reply .github/ISSUE_TEMPLATE/build_error.yml]: Good catch. Fixed using `an inline code block`

**Review by adamjstewart — DISMISSED** [review]

**Review by alalazo — DISMISSED** [review]

**Comment by gpotter2** [comment]
I just noticed it reads "Please upload the following files" but the file names are missing.

**Comment by gpotter2** [comment]
This should now be fixed.

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**